backend/infrahub/graphql/mutations/branch.py

```diff
@@ -129,18 +133,26 @@ class BranchUpdateInput(InputObjectType):
 class BranchDelete(Mutation):
     class Arguments:
         data = BranchNameInput(required=True)
+        wait_until_completion = Boolean(required=False)
 
     ok = Boolean()
+    task = Field(TaskInfo, required=False)
```

Comment:
I was wondering if we wanted required=True here to ensure that if a `task` is requested the returned data should be a `TaskInfo` instead of `Optional[TaskInfo]`. But looking at the rest of the code I think it might be correct. I.e. if a user requests the taskinfo to be returned that will only work if `wait_until_completion` is True otherwise task will not be returned. I think however that query might still break with the `return cls(ok=True)` and that it would need a `return cls(ok=True, task=None)`